plugins/LadspaEffect/calf/src/plugin.cpp

```diff
@@ -334,9 +334,13 @@ void ladspa_plugin_metadata_set::prepare(const plugin_metadata_iface *md, LADSPA
     param_count = md->get_param_count(); // XXXKF ladspa_instance<Module>::real_param_count();
     
     const ladspa_plugin_info &plugin_info = md->get_plugin_info();
+	std::string strName(plugin_info.name);
+	strName += " LAPSDA";
+	char *Name = new char[strName.size() + 1];
+	strcpy(Name, strName.c_str());
     descriptor.UniqueID = plugin_info.unique_id;
     descriptor.Label = plugin_info.label;
-    descriptor.Name = strdup((std::string(plugin_info.name) + " LADSPA").c_str());
```

Comment:
`strdup` is fine. You just need to save the result somewhere else. It is not as efficient, but correctness is more important.